### PR TITLE
add bid feature and show bids history if bids are available

### PR DESCRIPTION
**Pull request by agbanusi**

### CODE REVIEW

File: src/pages/NFTs/OpenBidNFT/BidModal.tsx [hunk 1]
```diff
@@ -181,20 +197,76 @@ const MESSAGE = styled.div`
...
+        })
+      }
+    })
+  }
+
+  const confirmBid = async () => {
+    setIsLoading(true)
+    const bidObject = {
+      clock: Date.now() + '', // string, not a number
+      tx_sig: 'RANDOM_TX_SIG_HERE',
```
Comment (agbanusi) [inline src/pages/NFTs/OpenBidNFT/BidModal.tsx:267]: The data needed for the bid post request is not complete but you can send a bid and receive a true response but the data doesn't update which I'm guessing the reason is the tokens and keys are not filled yet

File: src/pages/NFTs/OpenBidNFT/BidModal.tsx [hunk 2]
```diff
@@ -204,25 +276,66 @@ export const BidModal: FC<{ setVisible: (x: boolean) => void; visible: boolean }
...
+              <Col>
+                <img className="m-icon" src={`${process.env.PUBLIC_URL}/img/assets/close-white-icon.svg`} alt="" />
+              </Col>
+            </Row>
+            <div>Please try again, if the error persists please contact support.</div>
+          </MESSAGE>
+        )
+      })
+    }
+    // // TODO: Fake API
```
Comment (agbanusi) [inline src/pages/NFTs/OpenBidNFT/BidModal.tsx:321]: I didn't remove this yet for testing purposes

File: src/pages/NFTs/OpenBidNFT/BidModal.tsx [hunk 3]
```diff
@@ -204,25 +276,66 @@ export const BidModal: FC<{ setVisible: (x: boolean) => void; visible: boolean }
...
   return (
     <PURCHASE_MODAL setVisible={setVisible} title="" visible={visible} onCancel={onCancel}>
       <div className="bm-title">You are about to purchase a</div>
-      <Row className="bm-title" align="middle" gutter={4}>
-        <Col className="bm-title-bold">(Name of the NFT)</Col>
+      <Row className="bm-title" align="middle" justify="center" gutter={4}>
+        <Col className="bm-title-bold">{details?.nft_name || '(Name of the NFT)'}</Col>
         <Col>by</Col>
-        <Col className="bm-title-bold">(Name of the artist)</Col>.
+        <Col className="bm-title-bold">(Name of the artist).</Col>
```
Comment (agbanusi) [inline src/pages/NFTs/OpenBidNFT/BidModal.tsx]: I couldn't figure where to get the name of artist, which I'm guessing is owner's name the nearest I could find is the mint_address

Comment (walt-1) [reply src/pages/NFTs/OpenBidNFT/BidModal.tsx]: For future reference, its `nftMetadata.properties.creators` from the `useNFTDetails` context

File: src/pages/NFTs/OpenBidNFT/BidModal.tsx [hunk 4]
```diff
@@ -181,20 +197,76 @@ const MESSAGE = styled.div`
...
+      auction_house_treasury_mint_key: 'AUCTION_HOUSE_TREASURY_KEY_HERE',
+      token_account_mint_key: 'TOKEN_ACCOUNT_MINT_KEY_HERE',
+      buyer_price: 1.03 * Number(bidPrice) + '',
+      token_size: 'TOKEN_SIZE_HERE',
+      non_fungible_id: details.non_fungible_id,
+      collection_id: details.collection_id,
+      user_id: sessionUser?.user_id
+    }
+
+    const res = await bidOnSingleNFT(bidObject)
```
Comment (walt-1) [inline src/pages/NFTs/OpenBidNFT/BidModal.tsx]: For future reference, it is always good to wrap these kinds of calls in a `try { await call() } catch(err) {}` pattern.

**Pull request merged**